modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderOpenAPIResourceTest.java

```diff
@@ -759,59 +605,190 @@ private void _addAPIApplication() throws Exception {
 				"title", "title"
 			).toString(),
 			"headless-builder/applications", Http.Method.POST);
+
 		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SCHEMA_ERC, "/requestAPISchemaToAPIEndpoints/",
-				_API_POST_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SITE_SCOPED_SCHEMA_ERC, "/requestAPISchemaToAPIEndpoints/",
-				_API_SITE_SCOPED_POST_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SITE_SCOPED_SCHEMA_ERC,
-				"/responseAPISchemaToAPIEndpoints/",
-				_API_SITE_SCOPED_POST_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SCHEMA_ERC, "/responseAPISchemaToAPIEndpoints/",
-				_API_GET_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SINGLE_ELEMENT_SCHEMA_ERC,
-				"/responseAPISchemaToAPIEndpoints/",
-				_API_SINGLE_ELEMENT_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SINGLE_ELEMENT_SITE_SCOPED_SCHEMA_ERC,
-				"/responseAPISchemaToAPIEndpoints/",
-				_API_SINGLE_ELEMENT_SITE_SCOPED_ENDPOINT_ERC),
-			Http.Method.PUT);
-		assertSuccessfulJSONObject(
-			null,
-			StringBundler.concat(
-				"headless-builder/schemas/by-external-reference-code/",
-				_API_SITE_SCOPED_SCHEMA_ERC,
-				"/responseAPISchemaToAPIEndpoints/",
-				_API_SITE_SCOPED_ENDPOINT_ERC),
-			Http.Method.PUT);
+			JSONUtil.put(
```

Comment:
Could we do this in the same request?

Reply:
I mean by including it after the schema relationship name

Reply:
That's our current problem. If I do it that way we would get a NOT_FOUND because the schema is not created yet.